### ISSUE TITLE
[BUG] BCC error with list comprehension 

### ISSUE THREAD

**Issue by AlecHaring** [dashingsoft/pyarmor#1088]
**Executed command:** `pyarmor -d gen --enable-bcc .\pyarmor_test\`

**main.py**
```python
def testing():
    print("-".join(str(x) for x in range(3)))


if __name__ == "__main__":
    testing()

```

**pyarmor.error.log**
```
12872 MainProcess 2023-04-04 19:38:31,440
Traceback (most recent call last):
  File "C:\Users\alec\Desktop\PythonProjects\pyarmor-test\venv\lib\site-packages\pyarmor\cli\__main__.py", line 627, in main
    main_entry(sys.argv[1:])
  File "C:\Users\alec\Desktop\PythonProjects\pyarmor-test\venv\lib\site-packages\pyarmor\cli\__main__.py", line 615, in main_entry
    args.func(ctx, args)
  File "C:\Users\alec\Desktop\PythonProjects\pyarmor-test\venv\lib\site-packages\pyarmor\cli\__main__.py", line 179, in cmd_gen
    builder.process(options, pack=args.pack)
  File "C:\Users\alec\Desktop\PythonProjects\pyarmor-test\venv\lib\site-packages\pyarmor\cli\generate.py", line 158, in process
    self._obfuscate_scripts()
  File "C:\Users\alec\Desktop\PythonProjects\pyarmor-test\venv\lib\site-packages\pyarmor\cli\generate.py", line 120, in _obfuscate_scripts
    code = Pytransform3.generate_obfuscated_script(self.ctx, r)
  File "C:\Users\alec\Desktop\PythonProjects\pyarmor-test\venv\lib\site-packages\pyarmor\cli\core\__init__.py", line 48, in generate_obfuscated_script
    return m.generate_obfuscated_script(ctx, res)
  File "<maker>", line 760, in generate_obfuscated_script
  File "<maker>", line 26, in process
  File "<maker>", line 550, in process
  File "<maker>", line 550, in <listcomp>
  File "<bccmaker>", line 879, in process
  File "<bccmaker>", line 835, in trace
TypeError: not all arguments converted during string formatting

```

**Comment by jondy** [comment]
Thanks, reproduced and fixed in next release.

**Comment by jondy** [comment]
Fixed it in `8.1.dev5`
